Filebeat input httpjson getRFC5988Link parsing failure

The `getRFC5988Link` value template function will fail to parse Link strings in which there is no whitespace between the link's elements. As far as I understand the [spec](https://www.rfc-editor.org/rfc/rfc5988), whitespace is not required and there is at least one vendor producing pagination links without any whitespace. 

For confirmed bugs, please report:
- Version: Filebeat 8.6.1 (x-pack) and prior versions
- Operating System: Any
- Discuss Forum URL:
- Steps to Reproduce: 

# Alter an existing test case removing whitespace

```
$ git diff value_tpl_test.go
diff --git a/x-pack/filebeat/input/httpjson/value_tpl_test.go b/x-pack/filebeat/input/httpjson/value_tpl_test.go
index c5332a9658..b989e2128b 100644
--- a/x-pack/filebeat/input/httpjson/value_tpl_test.go
+++ b/x-pack/filebeat/input/httpjson/value_tpl_test.go
@@ -214,8 +214,8 @@ func TestValueTpl(t *testing.T) {
                                lastResponse: newTestResponse(
                                        nil,
                                        http.Header{"Link": []string{
-                                               `<https://example.com/api/v1/users?after=00ubfjQEMYBLRUWIEDKK>; title="Page 3"; rel="next"`,      
-                                               `<https://example.com/api/v1/users?before=00ubfjQEMYBLRUWIEDKK>; title="Page 1"; rel="previous"`, 
+                                               `<https://example.com/api/v1/users?after=00ubfjQEMYBLRUWIEDKK>;title="Page 3";rel="next"`,        
+                                               `<https://example.com/api/v1/users?before=00ubfjQEMYBLRUWIEDKK>;title="Page 1";rel="previous"`,   
                                        }},
                                        "",
                                ),
```

# Run test cases

```
#/go/src/github.com/elastic/beats/x-pack/filebeat/input/httpjson# go test -v . -run TestValueTpl/func_getRFC5988Link_multiple_rel_as_separate_strings_matches
=== RUN   TestValueTpl
=== RUN   TestValueTpl/func_getRFC5988Link_multiple_rel_as_separate_strings_matches
    value_tpl_test.go:680:
                Error Trace:    /go/src/github.com/elastic/beats/x-pack/filebeat/input/httpjson/value_tpl_test.go:680
                Error:          Not equal:
                                expected: "https://example.com/api/v1/users?before=00ubfjQEMYBLRUWIEDKK"
                                actual  : ""

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -https://example.com/api/v1/users?before=00ubfjQEMYBLRUWIEDKK
                                +
                Test:           TestValueTpl/func_getRFC5988Link_multiple_rel_as_separate_strings_matches
    value_tpl_test.go:682:
                Error Trace:    /go/src/github.com/elastic/beats/x-pack/filebeat/input/httpjson/value_tpl_test.go:682
                Error:          Received unexpected error:
                                the template result is empty
                Test:           TestValueTpl/func_getRFC5988Link_multiple_rel_as_separate_strings_matches
--- FAIL: TestValueTpl (0.00s)
    --- FAIL: TestValueTpl/func_getRFC5988Link_multiple_rel_as_separate_strings_matches (0.00s)
FAIL
FAIL    github.com/elastic/beats/v7/x-pack/filebeat/input/httpjson      0.008s
FAIL
```

# Potential Fix

https://github.com/elastic/beats/blob/37ddd1e4daa5c5679cf49a846b3f80d315966a24/x-pack/filebeat/input/httpjson/value_tpl.go#L231

```
-var regexpLinkRel = regexp.MustCompile(`<(.*)>;.*\srel\="?([^;"]*)`)
+var regexpLinkRel = regexp.MustCompile(`<(.*)>;.*rel\="?([^;"]*)`)
```

--- a/CHANGELOG.next.asciidoc
+++ b/CHANGELOG.next.asciidoc
@@ -120,6 +120,7 @@ automatic splitting at root level, if root level element is an array. {pull}3415
 
 - Allow the `misp` fileset in the Filebeat `threatintel` module to ignore CIDR ranges for an IP field. {issue}29949[29949] {pull}34195[34195]
 - Remove incorrect reference to CEL ext extensions package. {issue}34610[34610] {pull}34620[34620]
+- Fix handling of RFC5988 links' relation parameters by `getRFC5988Link` in HTTPJSON. {issue}34603[34603] {pull}34622[34622]
 
 *Auditbeat*
 

--- a/x-pack/filebeat/input/httpjson/value_tpl.go
+++ b/x-pack/filebeat/input/httpjson/value_tpl.go
@@ -228,7 +228,7 @@ func parseTimestampNano(ns int64) time.Time {
 	return time.Unix(0, ns).UTC()
 }
 
-var regexpLinkRel = regexp.MustCompile(`<(.*)>;.*\srel\="?([^;"]*)`)
+var regexpLinkRel = regexp.MustCompile(`<(.*)>.*;\s*rel\=("[^"]*"|[^"][^;]*[^"])`)
 
 func getMatchLink(rel string, linksSplit []string) string {
 	for _, link := range linksSplit {
@@ -241,7 +241,11 @@ func getMatchLink(rel string, linksSplit []string) string {
 			continue
 		}
 
-		if matches[2] != rel {
+		linkRel := matches[2]
+		if len(linkRel) > 1 && linkRel[0] == '"' { // We can only have a leading quote if we also have a separate trailing quote.
+			linkRel = linkRel[1 : len(linkRel)-1]
+		}
+		if linkRel != rel {
 			continue
 		}
 

--- a/x-pack/filebeat/input/httpjson/value_tpl_test.go
+++ b/x-pack/filebeat/input/httpjson/value_tpl_test.go
@@ -302,6 +302,24 @@ func TestValueTpl(t *testing.T) {
 			expectedVal: "https://example.com/default",
 		},
 		{
+			name:  "func getRFC5988Link no space link parameters",
+			value: `[[ getRFC5988Link "next" .last_response.header.Link ]]`,
+			paramCtx: &transformContext{
+				firstEvent: &mapstr.M{},
+				lastEvent:  &mapstr.M{},
+				lastResponse: newTestResponse(
+					nil,
+					http.Header{"Link": []string{
+						`<https://example.com/api/v1/users?before=00ubfjQEMYBLRUWIEDKK>;title="Page 1";rel="previous",
+						<https://example.com/api/v1/users?after=00ubfjQEMYBLRUWIEDKK>;title="Page 3";rel="next"`,
+					}},
+					"",
+				),
+			},
+			paramTr:     transformable{},
+			expectedVal: "https://example.com/api/v1/users?after=00ubfjQEMYBLRUWIEDKK",
+		},
+		{
 			name:        "can execute functions pipeline",
 			setup:       func() { timeNow = func() time.Time { return time.Unix(1604582732, 0).UTC() } },
 			teardown:    func() { timeNow = time.Now },

